stdext/string.ml

```diff
@@ -42,7 +42,7 @@ let drop_prefix s ~prefix =
 
 let drop_suffix s ~suffix =
   if is_suffix s ~suffix then
-    if length s = length suffix then Some s
+    if length s = length suffix then Some ""
```

Comment:
I am wondering whether there was some reason to keep it that (odd) way, since it specifically tests for this case.

After all, if you leave that check out you'd get exactly `Some ""` via `sub s ~pos:0 ~len:0`.

Reply:
That's true, but `drop_prefix` also has that check and it returns `""` if the prefix matches the string. I assumed the check was an optimization. The implementation of `String.drop_suffix` in dune's alternate stdlib also used to return the string when it matched the suffix but I changed it a few months ago to return `""`. The only place where `drop_suffix` is called is in dev_repo.ml, so no other code will be affected by this change. I propose we keep my change as it's now consistent with `drop_prefix` and also I think consistent with how most people will expect it to behave (or use `Base.String` and remove this function altogether).

Reply:
I've switched to using `Base.String.chop_suffix` and `chop_prefix` and removed the now unused `String.drop_suffix`